app/support/reservations/validations.rb

```diff
@@ -21,10 +21,22 @@ module Reservations::Validations
 
     validate :starts_before_ends
     validate :duration_is_interval
+    validate :category_is_valid
   end
 
   # Validation Methods
 
+  def category_is_valid
+    return if category.blank?
+    if self.class.const_defined?(:CATEGORIES)
```

Comment:
I'm not sure this is necessarily better, but what about something like (not fully tested)

```ruby
validates :category, inclusion: { in: -> (r) { r.class::CATEGORIES }, allow_blank: true },
      if: -> (r) { r.class.const_defined?(:CATEGORIES) }
 validates :category, absence: true, unless: -> (r) { r.class.const_defined?(:CATEGORIES) }
```